### PR TITLE
[SW-1445] Remove Deprecated setters on algorithms which has enum as argument

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala [hunk 1]
```diff
@@ -60,7 +59,9 @@ class H2OGridSearch(override val uid: String) extends Estimator[H2OMOJOModel]
   private var gridMojoModels: Array[H2OMOJOModel] = _
 
   override def fit(dataset: Dataset[_]): H2OMOJOModel = {
-    val algoParams = ${gridAlgoParams}
+    val algoParams = $ {
```
Comment (mn-mikke) [inline ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: Are we going to use this formating from now? What is negative about `val algoParams = ${gridAlgoParams}`?

Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: I actually prefer the original one, Intelij auto-formated this like this. I will try to see what we can do about this. Thanks for catching this!

Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: no problem

Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala]: Ah, I have found the cause why it was formatted like this. Actually we should use the parenthesis like this `$(gridAlgoParams)`

**Pull request merged**